src/components/settings-table/dataset-table.js

```diff
@@ -1,126 +1,59 @@
-import React from 'react'
-
-import Table from '@material-ui/core/Table'
-import TableBody from '@material-ui/core/TableBody'
-import TableCell from '@material-ui/core/TableCell'
-import TableHead from '@material-ui/core/TableHead'
-import TableRow from '@material-ui/core/TableRow'
+import React, { useEffect, useState } from 'react'
+import Wrapper from '../wrapper'
 import DataSetTableRow from './dataset-table-row'
-
-import i18n from '@dhis2/d2-i18n'
-import dataTableStyles from '../../styles/DataTable.module.css'
 import { periodTypeConstants } from '../../constants/data-set-settings'
 
-class DataSetTable extends React.Component {
-    constructor(props) {
-        super(props)
-
-        this.state = {
-            dataSetSelected: this.props.dataSetSelected,
-            dataSetFilter: '',
-            periodType: '',
-            defaultValue: '',
-            selected: false,
-        }
-    }
-
-    checkPeriodType = () => {
-        if (this.props.dataSetSelected !== '') {
-            const dataSetSelected = this.props.completeListOptions.filter(
-                dataSet => dataSet.id === this.props.dataSetSelected
+const DataSetTable = props => {
+    const [dataSet, setDataset] = useState(props.dataSetSelected)
+    const [dataSetData, setData] = useState(props.data)
+    const [periodType, setPeriodType] = useState('')
+    const [defaultValue, setDefault] = useState('')
+    const [dataSetFilter, setDataSetFiltered] = useState('')
+    const [selected, setSelected] = useState(false)
```

Comment:
It's better to use a single state object (`useState({ dataSet, periodType, ...etc })`) than a bunch of useStates when the state pieces are related